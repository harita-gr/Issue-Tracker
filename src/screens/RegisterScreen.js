
import React,{useEffect} from 'react'
import {Link,useHistory} from 'react-router-dom';
import {Row,Col,Form, Button, Container,Alert} from 'react-bootstrap'
import {useFormik} from 'formik'
import * as Yup from 'yup'
import { useDispatch,useSelector } from 'react-redux'
import { registerUser } from '../redux/actions/userActions'


function RegisterScreen() {

    const dispatch = useDispatch()
    const history = useHistory();

    const userRegister = useSelector((state) => state.userRegister)
    var { loading, error, userInfo } = userRegister

    useEffect(() => {
        window.scrollTo(0, 0)
      }, [error,userInfo])
      

        const formik = useFormik({
            initialValues:{
                email:'',
                password:'',
                confirmPassword:'',
                firstname:'',
                lastname:'',
                location:'',
                mobile:''
            },
    
            validationSchema:Yup.object().shape({
                email: Yup.string()
                            .required('Email is required'),
                password: Yup.string()
                            .required('Password is required')
                            .min(8,'Mininum 8 characters required'),
                confirmPassword: Yup.string()
                            .required('Confirm password is required'),
                firstname: Yup.string()
                            .required('Firstname is required')
                            .min(2,'Mininum 2 characters required'),
                lastname: Yup.string()
                            .required('Lastname is required'),
                mobile: Yup.string()
                            .required('Phone number is required')
                            .min(10,'10 digits required'),
                location: Yup.string()
                            .required('Location is required'),           
            }),
    
            onSubmit: values => {
                dispatch(registerUser(values));
                // history.push('/signin')
            }
        });
    
        return (
       
                <Container>               
                    <Row>
                        { error &&
                        <Col sm={12} md={6} lg={4} className="pt-2" className="mx-auto"  >   
                        <Alert variant='danger' align="center">{error}</Alert>
                        </Col> 
                        }    
                        { userInfo &&
                        <Col sm={12} md={6} lg={4} className="pt-2" className="mx-auto"  >   
                        <Alert variant='success' align="center">Successfully Registered! <Link to="/signin">Login</Link> to Continue</Alert>
                        </Col> 
                        }
                    <Col  sm={12}><h3 align="center" className="pt-2">REGISTER WITH US</h3></Col>
                    <Col  sm={12} md={6} lg={4} className="mx-auto">
      
                            <Form onSubmit={formik.handleSubmit}>
                            <Form.Group controlId="formBasicFirstname">
                                <Form.Label>First Name</Form.Label>
                                <Form.Control type="text" 
                                            name="firstname"
                                            id="firstname"
                                            placeholder="Enter first name" 
                                            onChange={formik.handleChange}
                                            onBlur={formik.handleBlur}
                                            value={formik.values.firstname}
                                            />
                                {formik.touched.firstname && formik.errors.firstname ? (<span style={{color:'red',margin:'5px'}}> {formik.errors.firstname}</span>): null} 
                            </Form.Group>
                            <Form.Group controlId="formBasicLastname">
                                <Form.Label>Last Name</Form.Label>
                                <Form.Control type="text" 
                                            name="lastname"
                                            id="lastname"
                                            placeholder="Enter last name" 
                                            onChange={formik.handleChange}
                                            onBlur={formik.handleBlur}
                                            value={formik.values.lastname}
                                            />
                                {formik.touched.lastname && formik.errors.lastname ? (<span style={{color:'red',margin:'5px'}}> {formik.errors.lastname}</span>): null} 
                            </Form.Group>
                            <Form.Group controlId="formBasicEmail">
                                <Form.Label>Email address</Form.Label>
                                <Form.Control type="email" 
                                            name="email"
                                            id="email"
                                            placeholder="Enter email" 
                                            onChange={formik.handleChange}
                                            onBlur={formik.handleBlur}
                                            value={formik.values.email}
                                            />
                                {formik.touched.email && formik.errors.email ? (<span style={{color:'red',margin:'5px'}}> {formik.errors.email}</span>): null} 
                            </Form.Group>
                            <Form.Group controlId="formBasicEmail">
                                <Form.Label>Location</Form.Label>
                                <Form.Control type="text" 
                                            name="location"
                                            id="location"
                                            placeholder="Enter your location" 
                                            onChange={formik.handleChange}
                                            onBlur={formik.handleBlur}
                                            value={formik.values.location}
                                            />
                                {formik.touched.location && formik.errors.location ? (<span style={{color:'red',margin:'5px'}}> {formik.errors.location}</span>): null} 
                            </Form.Group>

                            <Form.Group controlId="formBasicEmail">
                                <Form.Label>Phone Number</Form.Label>
                                <Form.Control type="text" 
                                            name="mobile"
                                            id="mobile"
                                            placeholder="Enter Phone number" 
                                            onChange={formik.handleChange}
                                            onBlur={formik.handleBlur}
                                            value={formik.values.mobile}
                                            />
                                {formik.touched.mobile && formik.errors.mobile ? (<span style={{color:'red',margin:'5px'}}> {formik.errors.mobile}</span>): null} 
                            </Form.Group>
                            
    
                            <Form.Group controlId="formBasicPassword">
                                <Form.Label>Password</Form.Label>
                                <Form.Control type="password" 
                                            name="password"
                                            id="password"
                                            placeholder="Password"
                                            onChange={formik.handleChange}
                                            onBlur={formik.handleBlur}
                                            value={formik.values.password} />
                                 {formik.touched.password && formik.errors.password ? (<span style={{color:'red',margin:'5px'}}> {formik.errors.password}</span>): null} 
    
                            </Form.Group>
                            <Form.Group controlId="formBasicEmail">
                                <Form.Label>Confirm Password</Form.Label>
                                <Form.Control type="password" 
                                            name="confirmPassword"
                                            id="confirmPassword"
                                            placeholder="Enter password again" 
                                            onChange={formik.handleChange}
                                            onBlur={formik.handleBlur}
                                            value={formik.values.confirmPassword}
                                            />
                                {formik.touched.confirmPassword && formik.errors.confirmPassword ? (<span style={{color:'red',margin:'5px'}}> {formik.errors.confirmPassword}</span>): null} 
                            </Form.Group>
                            <Button variant="primary" type="submit">
                                Submit
                            </Button>
                        </Form>
                    </Col>
                    <Col sm={12}><h5  align="center" className="pt-2">Already having account? <Link to="/signin">SIGN IN</Link></h5></Col>
                 </Row>
               
                </Container>
                
    )
}

export default RegisterScreen
