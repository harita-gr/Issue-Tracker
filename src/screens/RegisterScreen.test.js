import RegisterScreen from './RegisterScreen'
import {BrowserRouter as Router} from 'react-router-dom'
import React from 'react';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render } from '@testing-library/react'


describe('<RegisterScreen/>', () => {

    let wrapper;
    const initialState = {userRegister: { userInfo: 'dummyData',error:"" }}
    const mockStore = configureStore()
    let store;
  

    beforeEach( () =>{
        store = mockStore(initialState)
        wrapper = render(<Provider store={store}><Router><RegisterScreen/></Router></Provider>); 
        //console.log(wrapper.debug())
    })

        it('render the correct form title',() => {
            const { getByText } = wrapper
            expect(getByText('REGISTER WITH US')).not.toBeNull()
        })

        it('render the register form correctly',() =>{      
            //const container = render(<Provider store={store}><Router><RegisterScreen/></Router></Provider>)
            expect(wrapper).toMatchSnapshot();
          })

          afterEach( () =>{
            jest.clearAllMocks();
          })

});