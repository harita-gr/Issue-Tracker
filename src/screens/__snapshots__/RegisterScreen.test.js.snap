// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterScreen/> render the register form correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          
          <div
            class="mx-auto col-lg-4 col-md-6 col-sm-12"
          >
            <div
              align="center"
              class="fade alert alert-success show"
              role="alert"
            >
              Successfully Registered! 
              <a
                href="/signin"
              >
                Login
              </a>
               to Continue
            </div>
          </div>
          <div
            class="col-sm-12"
          >
            <h3
              align="center"
              class="pt-2"
            >
              REGISTER WITH US
            </h3>
          </div>
          <div
            class="mx-auto col-lg-4 col-md-6 col-sm-12"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicFirstname"
                >
                  First Name
                </label>
                <input
                  class="form-control"
                  id="firstname"
                  name="firstname"
                  placeholder="Enter first name"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                >
                  Last Name
                </label>
                <input
                  class="form-control"
                  id="lastname"
                  name="lastname"
                  placeholder="Enter last name"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                >
                  Email address
                </label>
                <input
                  class="form-control"
                  id="email"
                  name="email"
                  placeholder="Enter email"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                >
                  Location
                </label>
                <input
                  class="form-control"
                  id="location"
                  name="location"
                  placeholder="Enter your location"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                >
                  Phone Number
                </label>
                <input
                  class="form-control"
                  id="mobile"
                  name="mobile"
                  placeholder="Enter Phone number"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  id="password"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="formBasicEmail"
                >
                  Confirm Password
                </label>
                <input
                  class="form-control"
                  id="confirmPassword"
                  name="confirmPassword"
                  placeholder="Enter password again"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
          <div
            class="col-sm-12"
          >
            <h5
              align="center"
              class="pt-2"
            >
              Already having account? 
              <a
                href="/signin"
              >
                SIGN IN
              </a>
            </h5>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        
        <div
          class="mx-auto col-lg-4 col-md-6 col-sm-12"
        >
          <div
            align="center"
            class="fade alert alert-success show"
            role="alert"
          >
            Successfully Registered! 
            <a
              href="/signin"
            >
              Login
            </a>
             to Continue
          </div>
        </div>
        <div
          class="col-sm-12"
        >
          <h3
            align="center"
            class="pt-2"
          >
            REGISTER WITH US
          </h3>
        </div>
        <div
          class="mx-auto col-lg-4 col-md-6 col-sm-12"
        >
          <form
            class=""
          >
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="formBasicFirstname"
              >
                First Name
              </label>
              <input
                class="form-control"
                id="firstname"
                name="firstname"
                placeholder="Enter first name"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
              >
                Last Name
              </label>
              <input
                class="form-control"
                id="lastname"
                name="lastname"
                placeholder="Enter last name"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
              >
                Email address
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                placeholder="Enter email"
                type="email"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
              >
                Location
              </label>
              <input
                class="form-control"
                id="location"
                name="location"
                placeholder="Enter your location"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
              >
                Phone Number
              </label>
              <input
                class="form-control"
                id="mobile"
                name="mobile"
                placeholder="Enter Phone number"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
              >
                Password
              </label>
              <input
                class="form-control"
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="formBasicEmail"
              >
                Confirm Password
              </label>
              <input
                class="form-control"
                id="confirmPassword"
                name="confirmPassword"
                placeholder="Enter password again"
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
        <div
          class="col-sm-12"
        >
          <h5
            align="center"
            class="pt-2"
          >
            Already having account? 
            <a
              href="/signin"
            >
              SIGN IN
            </a>
          </h5>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
